### PR TITLE
Refactor Share Buttons

### PR DESCRIPTION
**Pull request by redreceipt**
## DESCRIPTION

Kapture 2019-07-29 at 16 19 23

### What does this PR do, or why is it needed?

Currently sharing text features doesn't work, this fixes that. The `<Share>` implementation was confusing and not working like we expected it to be. This clarifies what it's doing a little better by changing the naming.

### What design trade-offs/decisions were made?

Removes pulling `message` from the API. We weren't using it anyway in the client. We can discuss later on expanding functionality.

### How do I test this PR?

- Share a piece of content
- Share a text feature (Gathering Series, Spur One Another has one)

### What automated tests did you write?

None.

## TODO:

- [x] PR has a relevant title that will be understandable in a public changelog (ie...non developers)
- [x] Closes [#934 ]
- [x] No new warnings in tests, in storybook, and in-app
- [x] Upload GIF(s) of iOS and Android
- [x] Set a relevant reviewer

## REVIEW:

**Manual QA**

- [ ] Manual QA on iOS and ensure it looks/behaves as expected
- [ ] Manual QA on Android and ensure it looks/behaves as expected

**Code Review: Questions to consider**

- [ ] Read through the "Files changed" tab _very carefully_
- [ ] Edge cases: what assumptions are made about input?
- [ ] What kind of tests could be written?
- [ ] How might we make this easier for someone else to understand?
- [ ] Could the code be simpler?
- [ ] Will the code be easy to modify in the future?
- [ ] What's one part of these changes that makes you excited to merge it?

> The purpose of PR Review is to _improve the quality of the software._

### CODE REVIEW

**Comment by vinnyjth** [comment]
@michael-riddering I like the direction this PR takes and I feel like it does a good job of cleaning this code up and adding some clarity. 

However, I don't think I understand why you are removing `message` altogether. What's the harm in keeping `message`, and understanding that right now we don't send a `message` over from the API?

**Comment by vinnyjth** [comment]
OH! I see that we don't pass `message` to the `Share.share` call. We should fix that and pass `message`.

**Comment by vinnyjth** [comment]
🤦‍♂ @redreceipt

**Comment by redreceipt** [comment]
the danger in passing it to the share call is that it behaves differently on iOS and Android, but `title` always behaves the same and does what we are expecting `message` to do. which is why we opted to do it like this I think. I think before we just pass `message`, we should define what we want to happen with the share function and then code accordingly.

and I'll always opt for removing complexity and revisiting when there's a need.

**Comment by richarddubay** [comment]
I'm kind of confused here. As best I can tell, according to the RN docs (https://facebook.github.io/react-native/docs/share) the only thing that is iOS only is `url`. `message` and `title` work everywhere. 

In fact, those docs say that one of `message` or `url` is required ... so that makes me believe that `message` actually needs to be there so that sharing would work on Android. Is that not the case?

I'm assuming this is why the code we currently have in `share.js` looks like this:

```
const share = ({ title, url }) => {
  Share.share({
    title,
    message: Platform.OS === 'android' ? `${title}\n${url}` : title,
    url,
  });
};
```

**Comment by redreceipt** [comment]
once #938 gets merged in, this PR will be simplified to just refactor some names

**Comment by redreceipt** [comment]
ok @vinnyjth I think this one still has merit

File: packages/apolloschurchapp/src/ui/ShareContentButtonConnected/getShareContent.js [hunk 1]
```diff
@@ -7,7 +7,6 @@ export default gql`
         id
         sharing {
           url
-          message
```
Comment (vinnyjth) [inline packages/apolloschurchapp/src/ui/ShareContentButtonConnected/getShareContent.js]: We need to roll back this change

File: packages/apolloschurchapp/src/content-single/ActionContainer/index.js [hunk 2]
```diff
@@ -23,7 +23,7 @@ const ActionContainer = ({ itemId }) => (
     <MediaPlayerSpacer>
       <PositioningView>
         <LikeButton itemId={itemId} />
```
Comment (vinnyjth) [inline packages/apolloschurchapp/src/content-single/ActionContainer/index.js]: Can you rename LikeButton in a similar way as a part of this PR? Their functionality is very similar and I would like them to match :)

File: packages/apolloschurchapp/src/ui/ShareContentButtonConnected/getShareContent.js (hunk 1 above)
Comment (redreceipt) [reply packages/apolloschurchapp/src/ui/ShareContentButtonConnected/getShareContent.js]: done

File: packages/apolloschurchapp/src/content-single/ActionContainer/index.js (hunk 2 above)
Comment (redreceipt) [reply packages/apolloschurchapp/src/content-single/ActionContainer/index.js]: done

**Pull request merged**